core/src/main/java/org/apache/calcite/sql/validate/TableNamespace.java

```diff
@@ -61,22 +61,19 @@ private TableNamespace(SqlValidatorImpl validator, SqlValidatorTable table,
   }
 
   @Override protected RelDataType validateImpl(RelDataType targetRowType) {
-    this.mustFilterFields = ImmutableBitSet.of();
     table.maybeUnwrap(SemanticTable.class)
-        .ifPresent(semanticTable ->
-            this.mustFilterFields =
-                table.getRowType().getFieldList().stream()
-                    .map(RelDataTypeField::getIndex)
-                    .filter(semanticTable::mustFilter)
-                    .collect(toImmutableBitSet()));
-    table.maybeUnwrap(SemanticTable.class)
-        .ifPresent(semanticTable ->
-            this.mustFilterBypassFields = semanticTable.bypassFieldList()
-                .stream().collect(toImmutableBitSet()));
-    table.maybeUnwrap(SemanticTable.class)
-        .ifPresent(semanticTable ->
-            this.remnantMustFilterFields = new HashSet<>(semanticTable.remnantMustFilterFields()));
-
+        .ifPresent(semanticTable -> {
+          ImmutableBitSet mustFilterFields = table.getRowType().getFieldList().stream()
+              .map(RelDataTypeField::getIndex)
+              .filter(semanticTable::mustFilter)
+              .collect(toImmutableBitSet());
+          ImmutableBitSet bypassFieldList = semanticTable.bypassFieldList().stream()
+              .collect(toImmutableBitSet());
+          // We pass in an empty set for remnantMustFilterFields here because it isn't exposed to
```

Comment:
```
semanticTable.bypassFieldList().stream()
              .collect(toImmutableBitSet())
```
could be `semanticTable.bypassFieldList()` or `ImmutableBitSet.of(semanticTable.bypassFieldList())`